src/models/client.js

```diff
@@ -0,0 +1,40 @@
+import uuid from 'node-uuid';
+import Debug from 'debug';
+
+let clientStore = [];
+
+export class Client {
+  constructor(socket) {
+    this.id = uuid.v4();
+    this.socket = socket;
+    this._receivers = [];
+    this.debug = Debug(`app:client:${this.id}`);
+
+    this.socket.on('message', this._receive);
+    this.socket.on('close', this._close);
+    clientStore.push(this);
+  }
+
+  _receive(message) {
+    const data = JSON.parse(message);
+    this._receivers.forEach(rc => rc(data));
+  }
+
+  _close() {
+    this.debug('socket connection closed');
+    clientStore = clientStore.filter(client => client.id !== this.id);
+  }
+
+  send(data) {
```

Comment:
Do we want to force them to send an object? Maybe the message type is simple enough and they only need to send a string.  This can be easily checked for.